src/components/pdf-recommender/PDFRecommender.jsx

```diff
@@ -7,6 +7,11 @@ const PDFRecommender = ({ containerWidth }) => {
   const [recommenderLoaded, setRecommenderLoaded] = useState(false)
 
   useEffect(() => {
+    // TODO: It has to be set in local storage. This is not a very secure way.
+    //  We should find another way I believe.
+    localStorage.setItem('idRecommender', 'ffffff')
+    localStorage.setItem('userInput', '{}')
```

Comment:
I would prefer such things in the single place like a persistence.

Writing into `localStorage` is the same as writing into `window` or `global`. Nobody is going to complain that writing into global space is bad.

(not a change request, just a question)

Reply:
Actually I don't think so. I prefer to handle this in components. It's the same as styles. But on the other hand, I think localStorage should be cleaned after component is unmounted. What do you think?


Why do you think a single place is better and how would you do it?

Let's have an imagined example. There is  button for switching background color of website. The only thing what should the button do is set `theme=DARK` or `theme=BRIGHT` in localStorage.

Would you export a function from persistence file which would encapsulate `localStorage.setItem('theme, 'DARK'|'BRIGHT')` or would you rather subscribe for some event and no function wouldn't be exported at all?